src/ai/backend/common/lock.py

```diff
@@ -250,7 +250,14 @@ async def __aenter__(self) -> None:
 
     async def __aexit__(self, *exc_info) -> Optional[bool]:
         assert self._lock is not None
-        val = await self._lock.__aexit__(*exc_info)  # type: ignore[func-returns-value]
+        try:
+            val = await self._lock.__aexit__(*exc_info)  # type: ignore[func-returns-value]
+        except LockNotOwnedError:
+            log.exception("Lock no longer owned. Skip.")
+            return None
+        except LockError:
+            log.exception("Already unlocked. Skip.")
+            return None
```

Comment:
If we want to suppress this exception, how about returning a true value?
ref: https://docs.python.org/3.11/reference/compound_stmts.html#the-with-statement

Reply:
Or, we should check if the lock is properly acquired first and skip the unlocking operation entirely in the `__aexit__()` handler if it's not acquired.

Reply:
checking the lock is acquired in Redis lock would break the abstraction.
IMO, returning `True` makes sense!